backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/lsm/LiveMigrateDiskCommand.java

```diff
@@ -214,9 +220,29 @@ protected CreateSnapshotForVmParameters getCreateSnapshotParameters() {
         params.setDiskIdsToIgnoreInChecks(getImageGroupIds());
         params.setNeedsLocking(false);
         params.setEndProcedure(EndProcedure.COMMAND_MANAGED);
+
+        updateImageInitialSizeInBytes(params);
+
         return params;
     }
 
+    private void updateImageInitialSizeInBytes(CreateSnapshotForVmParameters params) {
+        if (FeatureSupported.isLiveStorageMigrationInitialSizeSupported(getVm())) {
+            StorageDomainStatic sourceDomain = storageDomainStaticDao.get(getParameters().getSourceStorageDomainId());
```

Comment:
This looks wrong - what is the source storage domain? When creating snapshot per vm, we have multiple disks, and they can be on different storage domains.

This command have multiple image groups ids (getImageGroupIds()), so I think we should have a map with one entry per disk on block domain, with the initial size for this disk snapshot.

Something like this (sorry for the pseudo java):

    // Compute snapshot initial size using config.
    snapshotInitialSize = ...

    Map<Guid, DiskImage> diskImageMap = new HashMap<Guid, DiskImage>();

    for (disk in disks) {
        if (disk storage domain is block domain) {
            disk = new DiskImage();
            initialSize = Math.min(snapshotInitialSize, disk.getSize());
            disk.setInitialSizeInBytes(initialSize);
            diskImageMap.put(disk.id, disk);
    }